lib/diagnostic.js

```diff
@@ -20,7 +31,18 @@ const sum = function sum() {
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let newArr = [];
+  if(defaultValue instanceof Array) {
+    for(let i = 0; i < length; i++) {
+      newArr[i] = defaultValue[i];
+    }
+    return newArr;
+  }
+  else if (defaultValue instanceof Function) {
+    array.forEach(function(element, index, array) {
+    array[index] = callback(element,index, array);
```

Comment:
Why not just pass in the `defaultValue` function as the callback for `.forEach`?